content/blog/2020-03-dplyr-1-0-0-new-verb-features/index.Rmarkdown

```diff
@@ -0,0 +1,231 @@
+
+```{r}
+out <- df %>% 
+  group_by(grp) %>% 
+  summarise(y = quantile2(y, c(0.25, 0.75)))
+out
+```
+Look carefully at the output - you'll see a `$` in the column names. This is a suggestion that something weird is going on and you have what we call a **df-column** because you have a column of a data frame that is itself a data frame! You can confirm that by extracting just that column:
+
+```{r}
+out$y
+```
+
+And of course, you can dig still deeper to get the individual values:
+
+```{r}
+out$y$y
+```
+
+Df-columns are surprisingly important to the internals of dplyr 1.0.0, but you should able to ignore their existence unless you deliberately want to try them out. They're definitely an advanced topic, and are something that we're continuing to play around with. We're also thinking about how improve the tibble print method to make it more obvious that something unusual is going on.
+
+### Non-summaries
+
+In combination with [`rowwise()`](http://dplyr.tidyverse.org/dev/articles/rowwise.html) (more on that in a future blog post), `summarise()` is now sufficiently powerful enough to replace many workflows that previously required a `map()` or `apply()` function. For example, to read all the all the `.csv` files in the current directory, you could write:
+
+```{r, eval = FALSE}
+tibble(path = dir(pattern = "\\.csv$")) %>% 
+  rowwise(path) %>% 
+  summarise(read.csv(path))
+```
+
+I feel deeply ambivalent about this code: it seems rather forced to claim that `read.csv()` computes a summary of a file path, but it's rather elegant pattern for reading in many files into a tibble.
+
+## Improved tidy select syntax
+
+`select()` and `rename()` are now significantly more flexible thanks to enhancements to the [tidyselect](https://tidyselect.r-lib.org/) package.
+
+### Basics
+
+Tidy selection provides a domain specific language that provides five ways of selecting variables:
+
+* By **position**: `df %>% select(1, 5, 10)` or `df %>% select(1:4)`.
+  Selecting variables is not generally recommended, but `rename()`ing
+  variables based on position can be useful, particularly if the variable
+  names are very long, non-syntactic, or duplicated.
+ 
+* By **name**: `df %>% select(a, e, j)` or `df %>% select(a:d)`.
+
+* By **function of name**: `df %>% select(starts_with("x"))`, or
+  `df %>% select(ends_with("s"))`. You can also use helpers `contains()`
+  and `matches()` for more flexibly matching.
+
+* By **type**: `df %>% select(is.numeric)`, `df %>% select(is.factor)`.
+
+* By **any combination** of the above using Boolean algebra `!`, `&`, `|`:
+
+    * `df %>% select(!is.factor)`: selects all non-factor variables.
+    
+    * `df %>% select(is.numeric & starts_with("x"))`: selects all
+      numeric variables that starts with "x".
+      
+    * `df %>% select(starts_with("a") | ends_with("z"))`: selects all
+      variables that starts with "a" or ends with "z".
```

Comment:
Feels like at least one example would be good here.

Reply:
Yeah, agreed. I'm going to break this blog post up into three (summarise, select + relocate + mutate, misc) so there's a bit more room to breathe.